src/Mod/Part/App/CrossSection.h

```diff
@@ -52,6 +53,23 @@ class PartExport CrossSection
     const TopoDS_Shape& s;
 };
 
-}
+class PartExport TopoCrossSection

```

Comment:
Is there precedent for making this a separate class (instead of a element-map-aware constructor in the CrossSection class)? It feels out-of-pattern to me, but I'm not as close to the code as you are.

Reply:
The only precedent I'm aware of is that's how the TNP branch did this.  It is worth noting that while at first glance the methods look the same, there are changes buried in there.  I think it would be possible to consolidate this, but there may be a fair amount of work involved.  What I'm not positive of, but I suspect, is that the CrossSection section completely goes away and we just have the TopoCrossSection part after the big flip.